Object Hash Function

lzardy opened the pull request:
Replaces GetChampionHash in the Champion class with GetObjHash in the ObjAiBase class.
Meant to be for future ease of access when assigning PackageHash in packets.
Will likely need to be modified in the future to account for units besides Champions that have skins.

lzardy committed:
Object Hash Function

Replaces GetChampionHash in the Champion class with GetObjHash in the ObjAiBase class.
Meant to be for future ease of access when assigning PackageHash in packets.
Will likely need to be modified in the future to account for units besides Champions that have skins.

danil179 commented on GameServerLib/GameObjects/AttackableUnits/AI/ObjAIBase.cs:
```diff
@@ -676,5 +676,46 @@ public void SetDashingState(bool state)
+            for (var i = 0; i < gobj.Length; i++)
+            {
+                hash = char.ToLower(gobj[i]) + 0x1003F * hash;
+            }
```
These for loops are repetitive blocks of code that implement basic hashes. I'm sure that this "hashing" is already present in LS. Please use the already existing hash function (if it doesn't exist, create one).

lzardy replied:
If I create one, should I use these for loops?

danil179 reviewed (changes requested):
I have one small comment.

moonshadow565 commented:
The hash function itself is likely something that should be into core.

lzardy committed:
Re-implement GetObjHash

Adds the HashStringNorm HashFunction for general use in hashing single strings.
Re-implements GetObjHash using HashStringNorm.

danil179 reviewed (changes requested):
Please change the function `HashStringSdbm` that is located above your new function in `HashFunctions.cs` to use your new method.

lzardy committed:
Use HashStringNorm in HashStringSdbm

lzardy commented:
> Please change the function `HashStringSdbm` that is located above your new function in `HashFunctions.cs` to use your new method.

Resolved.

danil179 reviewed (approved):
Approved. Please open an issue describing the PR.

lzardy commented:
This PR fixes #952.
